diff --git a/libvirt-test-api b/libvirt-test-api
index c93bd38..1cc3465 100755
--- a/libvirt-test-api
+++ b/libvirt-test-api
@@ -21,6 +21,7 @@ import time
 import getopt
 import shutil
 import tempfile
+import commands
 
 from src import parser
 from src import proxy
@@ -141,7 +142,8 @@ class Main(object):
             testid = logname[-3:]
             log_xml_parser.add_test_xml(testrunid, testid)
             if os.environ.has_key('AUTODIR'):
-                autotest_testdir = os.path.join(os.environ['AUTODIR'],'tests/libvirt_test_API')
+                autotest_testdir = os.path.join(os.environ['AUTODIR'],'tests/libvirt_test_api')
+                commands.getoutput("mkdir -p %s/src/log/%s" % (autotest_testdir, testrunid))
                 logfile = os.path.join('%s/src/log/%s' % (autotest_testdir, testrunid), logname)
             else:
                 logfile = os.path.join('log/%s' % testrunid, logname)
